packages/blocks/azure/src/lib/actions/custom-azure-api-action.ts

```diff
@@ -0,0 +1,93 @@
+import { createCustomApiCallAction } from '@openops/blocks-common';
+import { Property } from '@openops/blocks-framework';
+import {
+  authenticateUserWithAzure,
+  azureAuth,
+  getUseHostSessionProperty,
+} from '@openops/common';
+import { runCommand } from '../azure-cli';
+import { getSubscriptionsDropdownForHostSession } from '../common-properties';
+
+const getHostAccessToken = async (
+  auth: unknown,
+  subscription: string,
+): Promise<string> => {
+  const output = await runCommand(
+    'account get-access-token --resource https://management.azure.com --output json',
+    auth,
+    true,
+    subscription,
+  );
+  const parsed = JSON.parse(output ?? '{}');
+  const token = parsed?.accessToken;
+  if (!token) {
+    throw new Error('Failed to obtain Azure access token');
+  }
+  return token as string;
+};
+
+export const customAzureApiCallAction = createCustomApiCallAction({
+  auth: azureAuth,
+  name: 'custom_azure_api_call',
+  description: 'Make a custom REST API call to Azure.',
+  displayName: 'Custom Azure API Call',
+  baseUrl: () => 'https://management.azure.com/?api-version=2025-04-01',
+  additionalProps: {
+    documentation: Property.MarkDown({
+      value:
+        'For more information, visit the [Azure API documentation](https://learn.microsoft.com/rest/api/azure/).',
+    }),
+    useHostSession: getUseHostSessionProperty('Azure', 'az login'),
+    subscriptions: Property.DynamicProperties({
+      displayName: '',
+      required: false,
+      refreshers: [
+        'auth',
+        'useHostSession',
+        'useHostSession.useHostSessionCheckbox',
+      ],
+      props: async ({
+        auth,
+        useHostSession,
+      }): Promise<{ [key: string]: any }> => {
+        try {
+          if (useHostSession?.['useHostSessionCheckbox'] as boolean) {
+            return {
+              subDropdown: await getSubscriptionsDropdownForHostSession(auth),
+            } as any;
+          }
+        } catch (error) {
+          return {
+            subDropdown: Property.StaticDropdown({
+              displayName: 'Subscriptions',
+              description: 'Select a single subscription from the list',
+              required: true,
+              options: {
+                disabled: true,
+                options: [],
+                placeholder: `Something went wrong fetching subscriptions`,
+                error: `${error}`,
+              },
+            }),
+          } as any;
+        }
+        return {};
```

Comment:
as a best practice for readability, it's best to short circuit and return this condition first

Reply:
personally i disagree, because we then have an inverted if, i think its harder to read that way. Now with the try catch moved i think this way is fine